WcaOnRails/app/models/post.rb

```diff
@@ -2,11 +2,37 @@
 
 class Post < ApplicationRecord
   belongs_to :author, class_name: "User"
+  has_many :post_tags, autosave: true, dependent: :destroy
 
   validates :title, presence: true, uniqueness: true
   validates :body, presence: true
   validates :slug, presence: true, uniqueness: true
 
+  def reload
+    @tags = nil
+    super
+  end
+
+  attr_writer :tags
+
+  def tags
+    @tags ||= post_tags.pluck(:tag).join(",")
```

Comment:
Wow. It took me half an hour to understand this code. Much longer to try to figure out how it should be.

I think what makes it so strange is that `tags` is treated like a field/column/attribute when it really isn't. So it becomes a misfit ghost, caught between worlds.

Complaining is easy, better ideas are harder: I think, but am not sure, that creating and deleting the tags directly from the controller is the better pattern.